### ISSUE TITLE
lots of prepositions and articles in the topics compared to 0.9.3

### ISSUE THREAD

**Issue by mmchowdhury9** [MaartenGr/BERTopic#502]
when titles of 50 scientific articles were used to find topics by using v0.9.3, there was no prepositions or article in the topics. However, in v0.9.4, with the same data set, there were a lot of them negating the relevance of the topics. I have attached the data files. 
[downing_refs.txt](https://github.com/MaartenGr/BERTopic/files/8461181/downing_refs.txt)
all default options were used with 'all-MiniLM-L6-v2' embedding model.

**Comment by mmchowdhury9** [comment]
I understand. But even after running the model many times in v0.9.3, I don't get prepositions or articles, whereas, in v0.9.4, I always get them!

**Comment by MaartenGr** [comment]
Could you share your entire code for running the model in v0.9.3 and v0.9.4? That way, I can take a closer look at what is happening.

**Comment by mmchowdhury9** [comment]
# Import packages
#python version 3.9.7
import random
from bertopic import BERTopic

# Create a BERTopic model
random.seed(42)
topic_model = BERTopic(embedding_model="all-MiniLM-L6-v2")

# reading the data file and preprocessing
with open('downing_refs.txt', 'r') as f:
    downing_refs_titles = []
    for line in f:
        downing_refs_titles.append(line.strip())
indices = range(len(downing_refs_titles))
texts = [downing_refs_titles[i] for i in indices]

# getting the topics from the model
topics, probs = topic_model.fit_transform(texts)
freq = topic_model.get_topic_info(); 

# getting the full topics
topics_sentences = []
nr_topics = topic_model.get_topic_info().shape[0]
i_range = freq['Topic'].to_list()
for i in i_range:
    single_topic_words = ''
    nr_words = len(topic_model.get_topic(i))
    for j in range(0, nr_words):
        if j < len(topic_model.get_topic(i)) - 1:
            single_topic_words += topic_model.get_topic(i)[j][0] + ' '
        else:
            single_topic_words += topic_model.get_topic(i)[j][0]+'.'
    topics_sentences.append(single_topic_words)
topics_sentences

**Comment by MaartenGr** [comment]
Alright, it took some digging but I found the issue! Between v0.9.3 and v0.9.4 I exposed the `diversity` parameter as a way to control how diverse you want your topic representation. In v0.9.3, this was automatically set at 0, which still gives some optimization with respect to the topic representation since even with 0, we are using MMR to select the best fitting words. 

However, in v0.9.4, I made it so that if diversity is set to `None`, MMR is not being used at all. In other words, I would advise using `diversity=0` to enable MMR which should result in the output you got previously. 

Having said that, I would strongly advise using BERTopic for at least a few hundred and preferably a few thousand documents. With 50 documents there is an argument to be made against using topic modeling since it is possible to label the documents yourself.

**Comment by mmchowdhury9** [comment]
Great that it is solved. Thanks. But I feel that the default value of diversity = 0 is a good choice rather than None. In my application scenario, I need to find topics automatically from even less than 50 records and I found the minimum is 10.  I also feel the argument is weak as even from 50 records, finding the topics manually is tiresome and not reasonable in this digital age. Could you suggest some other options to find topics automatically from less number of records?

**Comment by MaartenGr** [comment]
Glad to hear that the problem is solved! I can imagine that having the diversity at 0 works quite well for you. That is, however, not the case for every use case. By turning it off as a default, users can select themselves whether that is something they would like to be using. Moreover, it is necessary to supply BERTopic with an embedding model to do so, whilst that is not something that many users actually do as they use pre-computed embeddings. 

For less number of records, it might be worthwhile to combine BERTopic with [KeyphraseVectorizers](https://github.com/TimSchopf/KeyphraseVectorizers) and by making sure that HDBSCAN parameters are set low enough that it finds a sufficient amount of topics.

**Comment by mmchowdhury9** [comment]
Great to know the details. Now I can appreciate the choice. Thanks for your suggestions. I will check them out. Wish you a lovely day ahead.
